assembl/static2/js/app/graphql/VoteSession.graphql

```diff
@@ -10,58 +67,29 @@ query VoteSession($discussionPhaseId: Int!, $lang: String!) {
     instructionsSectionTitle(lang: $lang)
     instructionsSectionContent(lang: $lang)
     propositionsSectionTitle(lang: $lang)
-    modules: voteSpecifications {
-      ... on TokenVoteSpecification {
-        id
-        voteSessionId
-        titleEntries {
-          ...langStringEntry
-        }
-        instructionsEntries {
-          ...langStringEntry
-        }
-        exclusiveCategories
-        tokenCategories {
-          id
-          totalNumber
-          typename
-          titleEntries {
-            ...langStringEntry
-          }
-          color
-        }
-      }
-      ... on NumberGaugeVoteSpecification {
+    proposals {
+      ... on Idea {
         id
-        voteSessionId
+        title(lang: $lang)
+        description(lang: $lang)
         titleEntries {
           ...langStringEntry
         }
-        instructionsEntries {
+        descriptionEntries {
           ...langStringEntry
         }
-        minimum
-        maximum
-        nbTicks
-        unit
+        order
       }
-      ... on GaugeVoteSpecification {
-        id
-        voteSessionId
-        titleEntries {
-          ...langStringEntry
-        }
-        instructionsEntries {
-          ...langStringEntry
-        }
-        choices {
-          id
-          value
-          labelEntries {
-            ...langStringEntry
-          }
-        }
+      modules: voteSpecifications {
+        ...tokenVoteSpecification
+        ...numberGaugeVoteSpecification
+        ...gaugeVoteSpecification
       }
     }
+    modules: voteSpecifications {
```

Comment:
Why repeat these queries if one of the leaves (`proposal`) is fetching it?

Reply:
voteSession.modules are the templates only, those non associated to a proposal.
modules in a proposal are the modules associated to this proposal only.